package aerofs.inception.admin;
import "kvm.proto";

service AdminPanelService {
    rpc __error__(Void) returns(aerofs.inception.kvm.ErrorReply);

    // Get a list of VM host IDs attached to the manager.
    rpc GetVmHostIdsList(Void) returns(GetVmHostIdsListReply);

    // Forward a protobuf request to a specific VM host.
    rpc SendMsgToVmHost(SendMsgToVmHostCall) returns(SendMsgToVmHostReply);
}

message Void {}

message GetVmHostIdsListReply {
    repeated string vmHostIds = 1;
}

message SendMsgToVmHostCall {
    required string vmHostId = 1;
    required bytes data = 2;
}

message SendMsgToVmHostReply {
    required bytes data = 1;
}
